src/importnb/loader.py

```diff
@@ -143,57 +168,67 @@ def nodes_to_code(self, nodes, path="<unknown>", *, _optimize=-1):
             dont_inherit=True,
             optimize=_optimize,
         )
+        return code
 
-    def source_to_code(self, source, path="<unknown>", *, _optimize=-1):
+    def source_to_code(  # type: ignore[override]
+        self, source: str, path: str = "<unknown>", *, _optimize: int = -1
+    ) -> CodeType:
         """Tangle python source to compiled code by:
-        1. parsing the source as ast nodes
-        2. compiling the ast nodes as python code
+        1. parsing the source as AST nodes
+        2. compiling the AST nodes as python code
         """
         nodes = self.source_to_nodes(source, path, _optimize=_optimize)
         return self.nodes_to_code(nodes, path, _optimize=_optimize)
 
-    def get_data(self, path):
-        """get_data injects an input transformation before the raw text.
+    def get_data(self, path: str) -> bytes:
```

Comment:
the `-> bytes` (and handling it at various places) was critical for some downstreams, such as `sphinx-autodoc`, to be able to retrieve the source.